context/param/methodparams.go

```diff
@@ -0,0 +1,69 @@
+package param
+
+import (
+	"fmt"
+	"strings"
+)
+
+//MethodParam keeps param information to be auto passed to controller methods
+type MethodParam struct {
+	name     string
+	location paramLocation
```

Comment:
Don't know.. maybe it will confuse that this is the go type of the param (like int, float etc.)? 
Also I see swagger ref you linked also treats this as "location". 
I'm OK with both options: What do you decide?